crates/host/src/wasmbus/mod.rs

```diff
@@ -774,156 +802,28 @@ impl Bus for Handler {
-                        format!("wasmbus.rpc.{lattice}.{}", invocation.target.public_key)
-                    }
-                };
-
-                let timeout = needs_chunking.then_some(CHUNK_RPC_EXTRA_TIME); // TODO: add rpc_nats timeout
-                let request = async_nats::Request::new()
-                    .payload(payload.into())
-                    .timeout(timeout)
-                    .headers(headers); // TODO: remove headers once all providers are built off the new SDK, which parses the trace context in the invocation
-                let res = nats
-                    .send_request(topic, request)
-                    .await
-                    .context("failed to call provider")
-                    .map_err(|e| e.to_string())?;
-
-                let InvocationResponse {
-                    invocation_id,
-                    mut msg,
-                    content_length,
-                    error,
-                    ..
-                } = rmp_serde::from_slice(&res.payload)
-                    .context("failed to decode invocation response")
-                    .map_err(|e| e.to_string())?;
-                if invocation_id != invocation.id {
-                    return Err("invocation ID mismatch".into());
-                }
-
-                let resp_length = usize::try_from(content_length)
-                    .context("content length does not fit in usize")
-                    .map_err(|e| e.to_string())?;
-                if resp_length > CHUNK_THRESHOLD_BYTES {
-                    msg = chunk_endpoint
-                        .get_unchunkified_response(&invocation_id)
-                        .await
-                        .context("failed to dechunk response")
-                        .map_err(|e| e.to_string())?;
-                } else if resp_length != msg.len() {
-                    return Err("message size mismatch".into());
-                }
-
-                if let Some(error) = error {
-                    Err(error)
-                } else {
-                    res_w
-                        .write_all(&msg)
-                        .await
-                        .context("failed to write reply")
-                        .map_err(|e| e.to_string())?;
-                    Ok(())
-                }
-            }
-            .boxed(),
-            Box::new(req_w),
-            Box::new(res_r),
-        ))
-    }
+        instance: &str,
+        name: &str,
+        params: Vec<wrpc_transport::Value>,
+    ) -> anyhow::Result<Vec<wrpc_transport::Value>> {
+        if let Some(TargetEntity::Wrpc(interface_target)) = target {
+            let prefix = format!("{}.{}", self.lattice, interface_target.id);
+            let wrpc_client =
+                wrpc_transport_nats::Client::new(self.nats.clone(), prefix.to_string());
+            let (result, _tx) = wrpc_client
+                // TODO: get types
+                .invoke_dynamic(instance, name, params, &[])
+                .await?;
```

Comment:
Also for @rvolosatovs ... I've been trying to consider how we actually get the types here that we expect from the function, I'm not sure how we would do it unless we had access to the component imports. I'm starting to lose some brain power, can you just confirm that I'm not thinking of this function completely wrong? (e.g., this function is called when we need to invoke a target over wRPC, not when the component is invoked)

Reply:
Yeah, we'll need the imports here, I'll plumb it through.

All of this is a side-effect of wRPC library not having "native" wasmtime support, which I decided to pause for now. Once that's done, we'll remove this functionality

Reply:
https://github.com/wasmCloud/wasmCloud/pull/1570